src/storage/btree.c

```diff
@@ -30599,8 +30604,8 @@ btree_key_delete_remove_object (THREAD_ENTRY * thread_p, BTID_INT * btid_int, DB
 		  logtb_find_current_mvccid (thread_p)))
 	  /* Cannot check if class OID is NULL. Get it in debug mode. */
 	  || OID_ISNULL (BTREE_DELETE_CLASS_OID (delete_helper))
-	  || lock_has_lock_on_object (BTREE_DELETE_OID (delete_helper), BTREE_DELETE_CLASS_OID (delete_helper),
-				      logtb_get_current_tran_index (), X_LOCK) > 0);
+	  || btree_check_locking_for_unique_operation (thread_p, BTREE_DELETE_CLASS_OID (delete_helper),
```

Comment:
For delete case, it is acceptable only if transaction is aborted.

Reply:
I'd like to separate it as http://jira.cubrid.org/browse/CBRD-23276